src/app/zap-templates/templates/app/attribute-size-src.zapt

```diff
@@ -70,26 +72,53 @@ uint16_t emberAfCopyList(ClusterId clusterId, EmberAfAttributeMetadata * am, boo
         {{#if isList}}
             case {{asHex code 4}}: // {{name}}
             {
-                entryLength = {{size}};
+                {{#unless (isString type)}}
+                entryLength  = {{size}};
                 if (((index - 1) * entryLength) > (am->size - entryLength))
                 {
                     ChipLogError(Zcl, "Index %l is invalid.", index);
                     return 0;
                 }
                 entryOffset = static_cast<uint16_t>(entryOffset + ((index - 1) * entryLength));
+                {{/unless}}
                 {{#if isStruct}}
                 // Struct {{chipType}}
                 {{chipType}} * entry = reinterpret_cast<{{chipType}} *>(write ? src : dest);
                 {{#chip_attribute_list_entryTypes}}
                 {{#if (isString type)}}
-                copyListMember(write ? dest : (uint8_t *)&entry->{{name}}, write ? (uint8_t *)&entry->{{name}} : src, write, &entryOffset, {{maxLength}}); // {{type}}
+                chip::ByteSpan * {{name}}Span = reinterpret_cast<chip::ByteSpan *>(&entry->{{name}}); // {{type}}
+                if (CHIP_NO_ERROR != (write ? WriteByteSpan(dest + entryOffset, {{size}}, {{name}}Span) : ReadByteSpan(src + entryOffset, {{size}}, {{name}}Span)))
+                {
+                    ChipLogError(Zcl, "Index %l is invalid. Not enough remaining space", index);
+                    return 0;
+                }
+                entryOffset = static_cast<uint16_t>(entryOffset + {{size}});
```

Comment:
The code was first written with descriptor cluster in mind where `List` basically contains raw types. As a result it assumes that you can easily target a particular element by index. `Struct` has inherited this behaviour in the sense that it has a reserved amount of spaces for bytespan (where the reserved amount of space is retrieved from the zap file).

You are correct that this is far from perfect, as it generates 2 problems. 
  * Lost space in the attribute store
  * Useless bytes transmitted over the network
 
`List[OCTET_STRING]` does not assume a particular size and I need to rewrite the templates to generate something similar for structs. But since the code was already in this shape I did not have rewritten in this particular PR.